aws/cloudformation/drone-stack.yml

```diff
@@ -0,0 +1,367 @@
+# - After successful stack creation, the Github OAuth app Homepage URL / Authorization callback URL must be updated to point to the new ELB.
+#   Alternatively, a route 53 A record alias can be created for the ELB, and the OAuth app URLs can be pointed to that.
+
+AWSTemplateFormatVersion: 2010-09-09
+Description: 'AWS CloudFormation template for running Drone.io on ECS. Recreates these steps: https://docs.drone.io/installation/github/multi-machine/'
+Parameters:
+  VpcId:
+    Type: String
+    Description: VpcId
+  SubnetIds:
+    Type: CommaDelimitedList
+    Description: Comma separated list of existing VPC Subnet Ids where ECS instances will run
+  AZs:
+    Type: CommaDelimitedList
+    Description: Comma separated list of availability zones where drone master ECS instances will run
+  NumWorkers:
+    Type: Number
+    Description: Initial number of drone workers (also the number of concurrent builds which can run)
+    Default: 2
+Resources:
+# ELB resources:
+# ELB / Target Group / Listener / ELB Security Group
+
+  ElbSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: ELB Allowed Ports
+      VpcId: !Ref VpcId
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 80
+          ToPort: 80
+          CidrIp: 0.0.0.0/0
+      Tags:
+        -
+          Key: Name
+          Value: !Sub "ECS ${AWS::StackName} - ALB SecurityGroup"
+
+  DroneMasterTargetGroup:
+    Type: AWS::ElasticLoadBalancingV2::TargetGroup
+    Properties:
+      Name: !Sub "TargetGroup-${AWS::StackName}"
+      VpcId: !Ref VpcId
+      Port: 80
+      Protocol: HTTP
+      Tags:
+        -
+          Key: Name
+          Value: !Sub "ECS ${AWS::StackName} - TargetGroup"
+
+  DroneMasterELB:
+    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+    Properties:
+      Name: !Sub "ELB-${AWS::StackName}"
+      SecurityGroups:
+        - !Ref ElbSecurityGroup
+      Subnets: !Ref SubnetIds
+      Tags:
+        -
+          Key: Name
+          Value: !Sub "ECS ${AWS::StackName} - ELB"
+
+  LoadBalancerListener:
+    Type: AWS::ElasticLoadBalancingV2::Listener
+    Properties:
+      LoadBalancerArn: !Ref DroneMasterELB
+      Port: 80
+      Protocol: HTTP
+      DefaultActions:
+        - Type: forward
+          TargetGroupArn: !Ref DroneMasterTargetGroup
+
+# Drone Master resources:
+# Security Group / EC2 Launch Template / EC2 Autoscaling Group
+# ECS cluster / ECS service / ECS task definition
+
+  DroneMasterEcsSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: ECS Allowed Ports
+      VpcId: !Ref VpcId
+      SecurityGroupIngress:
+        -
+          IpProtocol: tcp
+          FromPort: 1
+          ToPort: 65535
+          SourceSecurityGroupId: !Ref ElbSecurityGroup
+        -
+          IpProtocol: tcp
+          FromPort: 22
+          ToPort: 22
+          CidrIp: 0.0.0.0/0
+
+  DroneMasterECSCluster:
+    Type: AWS::ECS::Cluster
```

Comment:
- Is there a reason the two ECS services can't both draw from the same ECS cluster fed by instances from a single Auto Scaling Group? Is it because they have different scaling properties / instance sizes?
- Is there a reason to use EC2-instance-backed ECS clusters instead of Fargate to host these ECS ervices?

Reply:
* "Is it because they have different scaling properties / instance sizes?" Yea, that's what I was thinking. It probably also works to have one ECS cluster with both ASGs registered to it, and then placement constraints which ensure the correct instance types are used for the tasks, but I'm not sure what the benefit would be. LMK if you have a reason to do it a different way.
* The agents need to access the host's docker socket to start new containers for our builds, which isn't allowed in Fargate for security reasons. The master probably could run on fargate; I didn't give that a lot of thought. I think having it run on EC2 will work better overall though because the database will still persist on the host (via bind-mounted volume) if the container dies for whatever reason.